src/components/calcite-date-picker-day/calcite-date-picker-day.scss

```diff
@@ -194,34 +194,102 @@
   @apply rounded-none shadow-none;
 }
 
-:host(:hover[range-hover]:not([selected]).focused--end),
-:host(:hover[range-hover]:not([selected]).focused--start) .calcite--rtl {
+:host(:hover[end-of-range][range-hover][dir="ltr"]) .day-wrapper,
+:host(:hover[start-of-range][range-hover][dir="rtl"]) .day-wrapper {
```

Comment:
Can you use the `.calcite--rtl` class when direction is rtl - instead of having both the attribute selectors here?